File: android/src/main/java/com/evotor/egais/mapper/TransferToShop.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.evotor.egais.converter.QuantityBigDecimalConverter
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

internal fun createTransferToShop(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("uuid", cursor.getString(cursor.getColumnIndexOrThrow("UUID")))
    result.putString("owner", cursor.getString(cursor.getColumnIndexOrThrow("OWNER")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("transferNumber", cursor.getString(cursor.getColumnIndexOrThrow("NUMBER")))
    result.putString("transferDate", cursor.getString(cursor.getColumnIndexOrThrow("TRANSFER_DATE")))
    result.putString("note", cursor.getString(cursor.getColumnIndexOrThrow("NOTE")))
    result.putString("status", cursor.getString(cursor.getColumnIndexOrThrow("STATUS")))
    result.putString("rejectComment", cursor.getString(cursor.getColumnIndexOrThrow("REJECT_COMMENT")))
    result.putString("replyId", cursor.getString(cursor.getColumnIndexOrThrow("REPLY_ID")))
    return result
}

internal fun createTransferToShopPosition(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("TransferToShopUuid", cursor.getString(cursor.getColumnIndexOrThrow("TRANSFER_TO_SHOP_ID")))
    result.putString("identity", cursor.getString(cursor.getColumnIndexOrThrow("IDENTITY")))
    result.putString("productCode", cursor.getString(cursor.getColumnIndexOrThrow("PRODUCT_CODE")))
    result.putDouble("quantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("QUANTITY"))).toDouble())
    result.putString("informF2RegId", cursor.getString(cursor.getColumnIndexOrThrow("INFORM_F2_REG_ID")))
    result.putString("informF2MarkInfoJson", cursor.getString(cursor.getColumnIndexOrThrow("INFORM_F2_MARK_INFO_JSON")))
    result.putMap("productInfo", createProductInfo(cursor))
    return result
}

File: model/document/TransferToShop.js
import ProductInfo from "../dictionary/ProductInfo"
import {TransferStatus} from "../types"

/**
 * @class module:transfer.TransferToShop
 * @classdesc Акт передачи продукции со склада в торговый зал.
 * @param {string} uuid Уникальный идентификатор
 * @param {string} owner Кто подает документы
 * @param {?string} identity Идентификатор документа (клиентский)
 * @param {string} transferNumber Номер документа
 * @param {?Date} transferDate Дата составления
 * @param {?string} note Примечание
 * @param {module:transfer#TransferStatus} status Статус обработки документа
 * @param {?string} rejectComment Комментарий для отказа
 * @param {?string} replyId Уникальный идентификатор документа (присваивается УТМ); совпадает с идентификатором исходящего документа, который получили в ответе
 */
export default class TransferToShop {
    constructor(uuid: string,
                owner: string,
                identity: (string | null),
                transferNumber: string,
                transferDate: (Date | null),
                note: (string | null),
                status: TransferStatus,
                rejectComment: (string | null),
                replyId: (string | null)) {
        this.uuid = uuid;
        this.owner = owner;
        this.identity = identity;
        this.transferNumber = transferNumber;
        this.transferDate = transferDate;
        this.note = note;
        this.status = status;
        this.rejectComment = rejectComment;
        this.replyId = replyId;
    }
}

/**
 * @class module:transfer.TransferToShopPosition
 * @classdesc Позиция акта передачи продукции со склада в торговый зал.
 * @param {string} transferToShopUuid Уникальный идентификатор документа передачи
 * @param {string} identity Идентификатор позиции внутри акта
 * @param {string} productCode Регистрационный код продукции
 * @param {number} quantity Количество
 * @param {?string} informF2RegId Регистрационный номер раздела справки 2
 * @param {?string} informF2MarkInfoJson Информация о марках в формате JSON
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукте
 */
export class TransferToShopPosition {
    constructor(transferToShopUuid: string,
                identity: string,
                productCode: string,
                quantity: number,
                informF2RegId: (string | null),
                informF2MarkInfoJson: (string | null),
                productInfo: ProductInfo) {
        this.transferToShopUuid = transferToShopUuid;
        this.identity = identity;
        this.productCode = productCode;
        this.quantity = quantity;
        this.informF2RegId = informF2RegId;
        this.informF2MarkInfoJson = informF2MarkInfoJson;
        this.productInfo = productInfo;
    }

}